src/api/IBMiContent.ts

```diff
@@ -651,7 +651,9 @@ export default class IBMiContent {
   }
 
   async memberResolve(member: string, files: QsysPath[]): Promise<IBMiMember | undefined> {
-    const command = `for f in ${files.map(file => `/QSYS.LIB/${file.library.toUpperCase()}.LIB/${file.name.toUpperCase()}.FILE/${member.toUpperCase()}.MBR`).join(` `)}; do if [ -f $f ]; then echo $f; break; fi; done`;
+    // Escape names for shell
+    const fixed = files.map(f => ({name: f.name.replace(/([$\\])/g,'\\$1'), library: f.library.replace(/([$\\])/g,'\\$1')}));
+    const command = `for f in ${fixed.map(file => `/QSYS.LIB/${file.library.toUpperCase()}.LIB/${file.name.toUpperCase()}.FILE/${member.toUpperCase()}.MBR`).join(` `)}; do if [ -f $f ]; then echo $f; break; fi; done`;
```

Comment:
Isn't `member` going to be a problem if it contains a `$` as well?
In which case it may be faster to escape it everywhere at once.
```suggestion
    const command = `for f in ${files.map(file => `/QSYS.LIB/${file.library}.LIB/${file.name}.FILE/${member}.MBR`.replace(/([$\\])/g,'\\$1').toUpperCase()).join(` `)}; do if [ -f $f ]; then echo $f; break; fi; done`;
```

Reply:
Oh yeah... I am totally blind. Good spot

Reply:
Moved the file list into a new variable, purely for readability.